sdk/core/System.ClientModel/src/ModelReaderWriter/IJsonModel.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+
+namespace System.ClientModel.Primitives
+{
+    /// <summary>
+    /// Allows an object to control its own JSON writing and reading.
+    /// </summary>
+    /// <typeparam name="T">The type the model can be converted into.</typeparam>
+    public interface IJsonModel<out T> : IPersistableModel<T>
```

Comment:
I'm probably way late to the game here for naming stuff, but I'd have expected this to be `IJsonPersistable` or `IJsonPersistableModel` based on the base class name.

Reply:
It was agreed the shorter name is fine here only reason we added Persistable on the other is to avoid single word name `IModel`

Reply:
@jsquire, I agree with you on this.  @KrzysztofCwalina suggested we address this naming issue after beta but before GA.  The BCL team agreed about this one, and I think since these types are not end-user facing there is consensus that a longer name would be acceptable and more descriptive.

@m-nash, would it be possible to file an issue to track this with other pre-GA items?  We don't need to block the beta PR on it.

Reply:
These names were already agreed on but opened this up just in case https://github.com/Azure/azure-sdk-for-net/issues/40163